client/truncate_log.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# Usage: truncate_log.sh <size_threshold_mb> <log_files...>
+# Example: truncate_log.sh 10 /results/log1.log /results/log2.log
+
+set -euo pipefail
+
+if [ "$#" -lt 2 ]; then
+    echo "Usage: $0 <size_threshold_mb> <log_files...>"
+    exit 1
+fi
+
+SIZE_THRESHOLD_MB=$1
+shift
+HEAD_SIZE=$((SIZE_THRESHOLD_MB * 9/10 * 1024 * 1024))
+TAIL_SIZE=$((SIZE_THRESHOLD_MB * 1/10 * 1024 * 1024))
+
+for log_file in "$@"; do
+    if [ ! -f ${log_file} ]; then
```

Comment:
enclose variables that might have values with spaces in quotes
also use `[[` instead of `[`: it's way less error prone, faster, and we don't need portability to non-bash shells.
```
if [[ ! -f "${log_file}" ]]; then
```
https://google.github.io/styleguide/shellguide.html#quoting